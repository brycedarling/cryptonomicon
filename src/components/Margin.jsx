import React, { Component } from 'react';

class Margin extends Component {
  render() {
    return (
      <g
        className="margin"
        transform={`translate(${this.props.left}, ${this.props.top})`}
      >
        {this.props.children}
      </g>
    );
  }
}

export default Margin;
