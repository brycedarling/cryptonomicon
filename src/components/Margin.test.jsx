import React from 'react';
import renderer from 'react-test-renderer';
import Margin from './Margin';

describe('components', () => {
  describe('Margin', () => {
    it('renders correctly', () => {
      const tree = renderer.create((
        <Margin
          top={10}
          left={20}
          bottom={30}
          right={40}
        >
          hi mom
        </Margin>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
