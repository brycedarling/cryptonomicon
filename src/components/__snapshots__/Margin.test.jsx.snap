// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Margin renders correctly 1`] = `
<g
  className="margin"
  transform="translate(20, 10)"
>
  hi mom
</g>
`;
